### ISSUE TITLE
Raven get slow when have many notifications

### ISSUE THREAD

**Issue by otavioschwanck** [solus-project/budgie-desktop#1802]
## Budgie version
10.5

## Operating system
Solus 4

## Steps to reproduce the issue
In terminal, using zsh, run:
for i in {1..1000}; do notify-send hi; done

## Actual result
Raven get very very slow.

## Expected result
Raven managing better the notification, not loading all.

**Comment by JoshStrobl** [comment]
The alternative is to present a real-world usecase and not spam opening a 1000 notifications.

**Comment by otavioschwanck** [comment]
> The alternative is to present a real-world usecase and not spam opening a 1000 notifications.

With 30 it get slow already... Using Franz with messenger, slack, whatsapp, you get 100 notifications really fast.

**Comment by MatthewScholefield** [comment]
Yeah, I'm having the same issue actually. Looks like it's rendering all of the notifications when raven opens each time. Personally, I'd opt for a reasonable cap on the number of notifications (beyond that, old ones get discarded).

**Comment by Krutonium** [comment]
@JoshStrobl How is this for real world: I run into this _Daily_. Between Discord and Telegram, I get upwards of 4000 notifications a day. Please re-open this.

**Comment by arcooke** [comment]
Author of dupe above. 

> The alternative is to present a real-world usecase and not spam opening a 1000 notifications.

KDE Connect generates tons of notifications.  Every text message, every email on all of my email accounts, every app notification.  These pile up organically over time.  This is how I end up with thousands of them.  I don't check the sidebar notifications often, but it's useful to have them archived.   I submitted my thread yesterday, and I currently have 217 unread notifications.

**Comment by Krutonium** [comment]
> Author of dupe above.
> 
> > The alternative is to present a real-world usecase and not spam opening a 1000 notifications.
> 
> KDE Connect generates tons of notifications. Every text message, every email on all of my email accounts, every app notification. These pile up organically over time. This is how I end up with thousands of them. I don't check the sidebar notifications often, but it's useful to have them archived. I submitted my thread yesterday, and I currently have 217 unread notifications.

In a similar vein, I've had over 700 since I last cleared it - and that wasn't super long ago!

**Comment by arcooke** [comment]
Pinging @JoshStrobl to have this reopened, it's a legitimate issue.  It might not appear to be an problem for folks who shut down/reboot often.  They get cleared on reboot, but those of us who keep our PCs on 24/7, they do stack up quite fast.

**Comment by Krutonium** [comment]
![Slow](https://user-images.githubusercontent.com/3945538/104838252-bd9dd980-5887-11eb-91e4-56d1140769ee.png)
Reiterating that this is a legitimate problem. When it gets to this point it can take upwards of 30 seconds for Raven to open, and while it is, the rest of the desktop is locked up.

**Comment by JoshStrobl** [comment]
I'll be looking into improving the performance of the Notifications section in a future 10.5.x release, though we are limited by the performance deficiencies of the ListBox in GTK3. This will likely occur by doing the following:

1. Cap per-app notifications
2. Limit visible notifications per-app and put the rest in an expander.

**Comment by LuckySandwich** [comment]
Thank you.  It's not just Raven that slows down, the whole system slows down.  I usually notice that I need to clear my notifications when Firefox starts getting sluggish. 

Something like this might be nice as well: https://www.htmlelements.com/demos/listbox/lazy-loading/

**Comment by Krutonium** [comment]
Thank you very much.

**Comment by crvigh** [comment]
https://github.com/solus-project/budgie-desktop/issues/922 might be a related issue, where Raven hangs the entire desktop when it receives notification from Rhythmbox.

**Comment by JoshStrobl** [comment]
It's not remotely related...
